src/main/java/com/minecolonies/entity/ai/citizen/farmer/EntityAIWorkFarmer.java

```diff
@@ -21,550 +21,698 @@
+
+            if (newField != null && getOwnBuilding() != null)
+            {
+                getOwnBuilding().addFarmerFields(newField);
+            }
+        }
+    }
+
+    /**
+     * Checks if the farmer is ready to plant.
+     *
+     * @param currentField the field to plant.
+     * @return true if he is ready.
+     */
+    private boolean canGoPlanting(@NotNull final Field currentField, @NotNull final BuildingFarmer buildingFarmer)
+    {
+        if (currentField.getSeed() == null)
+        {
+            chatSpamFilter.talkWithoutSpam("entity.farmer.noSeedSet");
+            buildingFarmer.setCurrentField(null);
+            return false;
+        }
+
+        if (shouldTryToGetSeed)
+        {
+            final int slot = worker.findFirstSlotInInventoryWith(currentField.getSeed());
+            final ItemStack seeds = new ItemStack(currentField.getSeed());
+            if (slot != -1)
+            {
+                requestSeeds = false;
+            }
+            if (!walkToBuilding())
+            {
+                if (isInHut(seeds))
+                {
+                    requestSeeds = false;
+                    isInHut(seeds);
+                }
+                shouldTryToGetSeed = requestSeeds;
+                if (requestSeeds)
+                {
+                    chatSpamFilter.talkWithoutSpam("entity.farmer.NeedSeed", currentField.getSeed().getItemStackDisplayName(seeds));
+                }
+            }
+        }
+
+        return !shouldTryToGetSeed;
+    }
+
+    /**
+     * The main work cycle of the Famer.
+     * This checks each block, harvests, tills, and plants.
+     */
+    private AIState cycle()
+    {
+        @Nullable final BuildingFarmer buildingFarmer = getOwnBuilding();
+
+        if (buildingFarmer == null || checkForHoe() || buildingFarmer.getCurrentField() == null)
+        {
+            return AIState.PREPARING;
+        }
+
+        @Nullable final Field field = buildingFarmer.getCurrentField();
+
+        if (workingOffset != null)
+        {
+            final BlockPos position = field.getLocation().down().south(workingOffset.getZ()).east(workingOffset.getX());
+            // Still moving to the block
+            if (walkToBlock(position.up()))
+            {
+                return AIState.FARMER_WORK;
+            }
+
+            // harvest the block if able to.
+            if (harvestIfAble(position))
+            {
+                setDelay(getLevelDelay());
+            }
+        }
+
+        if (!handleOffsetHarvest(field))
+        {
+            resetVariables();
+            shouldDumpInventory = true;
+            field.setNeedsWork(false);
+            return AIState.IDLE;
+        }
+        return AIState.FARMER_WORK;
+    }
+
+    /**
+     * This (re)initializes a field.
+     * Checks the block above to see if it is a plant, if so, breaks it. Then tills.
+     */
+    private AIState initialize()
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of returns of this method 7, down to the maximum allowed 5. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1142)